src/systems/engine.cppm

```diff
@@ -31,4 +37,22 @@ export namespace ecs::systems::engine {
         if (display->get().window->is_input_active(abstractions::gui::inputs::down))
             pos.y += c.speed;
     }
+
+    constexpr void update_hitbox_position(ecs::components::position &pos, ecs::components::engine::hitbox &box) noexcept {
+        box.box.x = pos.x;
+        box.box.y = pos.y;
+    }
+
+    constexpr void collision(entity e, entity_container &ec, components::engine::hitbox &box) noexcept
```

Comment:
This system does not reset the `components::engine::hitbox::triggered` field if no collision is found, so you will always be colliding after colliding once, and that's quite a big flaw in the collision algorithm, I think.